### ISSUE TITLE
BUG: Python 3.13 development wheels not available 

### ISSUE THREAD

**Issue by lesteve** [pandas-dev/pandas#59372]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
Look at https://anaconda.org/scientific-python-nightly-wheels/pandas/files, there is no cp313 wheels. There were some a few days ago but it seems like you are removing the old wheel.

Issues seem to be related to Python 3.13 see [build log](https://github.com/pandas-dev/pandas/actions/runs/10173576695/job/28137917210) (not Python 3.13 free-threaded in particular just to be clear)
```

### Issue Description

In scikit-learn CI we have started relying on pandas development wheels for Python 3.13 free-threaded work. Probably this was a bit over-optimistic on our side, see <this> for more details.

### Expected Behavior

There are some Python 3.13 wheels

### Installed Versions

N/A

**Comment by mroeschke** [comment]
Yeah our wheel jobs depend on the unit tests 100% passing (this _could_ be relaxed but there's a tradeoff of releasing a non-functioning nightly wheel).

The 3.13 wheels test against a numpy nighties which I think changed something that broke these unit tests, but no one has had time to look into the failure.

**Comment by lesteve** [comment]
OK thanks for the answer!

Actually looking a bit more, sorry for the noise, you do have the old wheels cp313-313t wheels see [this](https://anaconda.org/scientific-python-nightly-wheels/pandas/files?page=3) (I only looked at the first page initially).

Now I need to figure out why it seems like `pip` does not find the wheel somehow and tries to install from source in our CI ...

**Comment by mroeschke** [comment]
I'll try to address the Python 3.13 test failures by the end of the week so more recent wheels can be uploaded. It slipped my mind that these failures also affected our Python 3.13 nightlies

**Comment by lesteve** [comment]
No pressure from the scikit-learn side on this, we are completely fine using an old wheel for some time.

Because the sdist was more recent than the wheel, the sdist was chosen by pip first. Using `pip install --only-binary :all:` is the right way to convince pip to only use wheels and I have opened https://github.com/scikit-learn/scikit-learn/pull/29598 to do just that.

**Comment by ogrisel** [comment]
Actually, since older builds seem to be garbage collected, there aren't any cp313t wheel for pandas dev anymore: https://github.com/scikit-learn/scikit-learn/issues/29610#issuecomment-2265355183

**Comment by lysnikolaou** [comment]
It seems that most (all?) of the failures are related to a recent NumPy commit that appeared in the nightlies on July 24th. The NumPy commit that introduced the breakage is https://github.com/numpy/numpy/commit/682cd040f0997ece83af51ffe0e3241242c7c1df:

```diff
682cd040f0997ece83af51ffe0e3241242c7c1df is the first bad commit
commit 682cd040f0997ece83af51ffe0e3241242c7c1df (HEAD)
Author: Sebastian Berg <sebastianb@nvidia.com>
Date:   Wed Jul 10 16:50:32 2024 +0200

    BUG,ENH: Fix generic scalar infinite recursion issues
    
    This reorganizes the avoidance of infinite recursion to happen
    in the generic scalar fallback, rather than attempting to do so
    (badly) in the scalarmath, where it only applies to longdouble
    to begin with.
    
    This was also needed for the follow up to remove special handling
    of python scalar subclasses.

 numpy/_core/src/multiarray/npy_static_data.c |   1 +
 numpy/_core/src/multiarray/npy_static_data.h |   1 +
 numpy/_core/src/multiarray/scalartypes.c.src | 311 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++-------------
 numpy/_core/src/umath/scalarmath.c.src       |  16 ++---
 4 files changed, 276 insertions(+), 53 deletions(-)
```

**Comment by mroeschke** [comment]
The pandas CI is finally green wrt unit tests so our nightly wheel upload is finally working again

https://github.com/pandas-dev/pandas/actions/runs/10295469960
https://anaconda.org/scientific-python-nightly-wheels/pandas/files

so closing this out
